packages/lib/src/vis/scatter/ScatterPoints.tsx

```diff
@@ -88,6 +90,15 @@ function ScatterPoints(props: Props) {
     invalidate();
   }, [color, dataGeometry, invalidate, position]);
 
+  const zoomVector = useThree((state) => state.camera.scale);
+
+  useFrame(() => {
+    const zoom =
+      Math.sqrt(zoomVector.x * zoomVector.x + zoomVector.y * zoomVector.y) /
+      Math.SQRT2;
+    setThreshold((size * zoom) / 2);
+  });
```

Comment:
There is maybe a way to combine the `useFrame` and the `useEffect` in a single hook ? What do you think ?